docs/gadget-devel/output.md

```diff
@@ -0,0 +1,40 @@
+---
+title: 'Displaying Output'
+sidebar_position: 620
+---
+
+By default, Inspektor Gadget allows you to display the output of each data
+source in all the output modes supported by the [CLI
+operator](../spec/operators/cli.md#output). However, depending on the
+[types](./gadget-intro.md#data-sources-types) of your data source, some output
+modes may not be appropriate. You can customize the supported output modes and
+default output mode for each data source by setting the
+[cli.supported-output-modes](../spec/operators/cli.md#clisupported-output-modes)
+and [cli.default-output-mode](../spec/operators/cli.md#clidefault-output-mode) annotations.
+
+Unless you have specific requirements, we recommend using the CLI operator's
+default configuration. However, for [Map
+Iterators](./gadget-intro.md#map-iterators) with a map value type of
+`gadget_histogram_slot__u32` or `gadget_histogram_slot__u64` (TODO: Add links),
+the output should be displayed as a chart, which the CLI operator doesn't
+support. In such cases, you can use the [Otel
+Metrics](../spec/operators/otel-metrics.md) operator to render the output as a
+histogram by annotating the data source with [metrics.print:
+"true"](../spec/operators/otel-metrics.md#metricsprint). The
+[profile_blockio](../gadgets/profile_blockio.mdx) gadget is an example of this.
+
+## Fields
+
+The fields to be displayed in the output when using the `columns`, `json`,
+`jsonpretty` or `yaml` output modes depend on the data source type:
+
+- Tracers: The fields are all the elements of the event `struct` specified when
+  defining the data source.
+- Map Iterators: The fields are all the elements of the key and value `struct`s
+  used to define the eBPF map to be iterated. As mentioned above, there is a
+  special case for the map iterators with a map value type of
+  `gadget_histogram_slot__u32` or `gadget_histogram_slot__u64`. In such cases,
```

Comment:
I think histograms are a special case because those are the only ones we support printing to the terminal, but I don't see any reason why not to support printing counters and gauges too. i.e. all cases where are metrics involved will be "special".

Reply:
Mmmm but counters and gauges can simply be printed as columns, JSON, or any other output mode supported by the CLI operator, they don't require a special treatment.

Reply:
Makes sense, we can print other kind of metrics as columns.